### DISCUSSION TITLE
any recent windows builds with frei0r support?

### DISCUSSION THREAD

**Discussion by junguler** [mpv-player/mpv#14211] [Q&A] [answered]
hi, i'm on an [old build](https://github.com/qwerty12/mpv-winbuild) of mpv that has frei0r support on windows, the repo owner was kind enough to include these filters in his builds but have since stopped maintaining it

i need frei0r because it has a filter called colortap which converts overly bright/white colors to grey and makes watching videos on night time easier on the eyes

**Comment by ruihe774** [comment]
mpv has no official build. Please report it to the corresponding third party.

**Reply by junguler** [reply]
i understand what you mean, i just wanted to know if there is such a build out there, which there is and i'm satisfied with the results

**Comment by Andarwinux** [comment, marked as answer]
My personal [build](https://github.com/Andarwinux/mpv-winbuild) came with frei0r but was untested.

**Reply by junguler** [reply]
it works, thank you, please keep maintaining your repo as long as it's possible

**Comment by richardpl** [comment]
frei0r - bad (overly-simplistic) library - bad overall design, code quality, and under-perfoming output quality and performance.
Better filters than colortap are already in libavfilter native filters, see lut3d filter and curves filter among many more.

**Reply by junguler** [reply]
i'm not really knowledgeable enough to know if what you said is correct about frei0r but i know i've been using it for a while now and it works for my needs

having said that i'm always open to trying new things and will investigate the filters you've mentioned so thank you, i appreciate it

**Reply by richardpl** [reply]
-vf curves=preset=darker
There are also many other ways to do it with lut3d filter, just need to locate .cube 3d LUT files you like.

**Reply by kasper93** [reply]
If using lut is an option, it would be more efficient to use `--target-lut` directly https://mpv.io/manual/master/#options-target-lut (`gpu-next` only)

**Reply by junguler** [reply]
thank you for these replies, i have seriously look into these filters because i spend a lot of time behind a computer during night and this will help a lot

**Comment by junguler** [comment]
ok i figured out how to use a lut3d.cube file to re-create colortap without always needing frei0r, i will post the steps here for anyone else who might be interested

1- go to this [website](https://sirserch.github.io/lut-creator-js/#), click `Download Hald`, `Hald 25 size` this will give you a normal lut3d png
2- use ffmpeg to apply colortap filter to it 
```ffmpeg -i Neutral_25.PNG -vf frei0r=colortap output.png```
3- upload the new `output.png` to the website by pressing `Convert to .CUbe`
4- place this newly made lut3d file to somewhere and link it inside your `input.conf` as a shortcut
```CTRL+c vf toggle lut3d="C:\programs\mpv-x86_64\portable_config\luts\colortap.cube" ; show-text "lut3d colortap"```

**Comment by richardpl** [comment]
Pointless waste of time on reinventing inefficient wheel, ffmpeg also have haldclutsrc filter, so no need to use any website tracking nonsense.
So once you apply colortap frei0r filter on haldclutsrc video, you just use haldclut filter with the clut png in 2nd stream.
No generating of LUTs files on any web site is needed.

**Reply by junguler** [reply]
could you actually write what i should put in mpv.conf or input.conf to achieve this? i've been searching all over the internet since you mentioned this but couldn't find how to do this ...

i can do this on ffplay no problem but mpv doesn't seem to want to play nice at all

**Reply by richardpl** [reply]
What have you tried?

**Reply by junguler** [reply]
```
ffplay video.mp4 -vf "movie=hald.png, [in] haldclut"
```
```
mpv video.mp4 --vf="lavfi=haldclut=hald.png"
mpv video.mp4 --vf="haldclut=hald.png"
mpv video.mp4 --vf="lavfi=[haldclut=hald.png]"
mpv video.mp4 --vf="movie=hald.png,[in]haldclut"
```

**Reply by richardpl** [reply]
try: `mpv video.mpv -vf "lavfi=[null[v];movie=hald.png[lut];[v][lut]haldclut]"`

**Reply by junguler** [reply]
it works, thank you

**Reply by richardpl** [reply]
you can increase lut entries from default 6 level to at least level 8 (512x512 image size) in haldclutsrc filter, if needed for "better" quality.

**Reply by junguler** [reply]
i see, good to know, now that it worked i'm sure to play around with it a lot more

**Reply by junguler** [reply]
hello, sorry to bother you again but i can't seem to do the same thing as i did in the terminal as a keybind in input.conf, if my hald.png file is not in the same folder as the video playing it will not work, i'm on windows btw

i've tried escaping the address with `\\` `\\\` `\\\\` `\/` i've also tried varies combinations of quotes or double quotes in the address, none of them seem to be working if the png file is not in the same folder

```
vf toggle lavfi='[null[v];movie=/C/path/hald.png[lut];[v][lut]haldclut]'
vf toggle lavfi='[null[v];movie=C:/path/hald.png[lut];[v][lut]haldclut]'
vf toggle lavfi='[null[v];movie=C\/:\/path\/hald.png[lut];[v][lut]haldclut]'
vf toggle lavfi='[null[v];movie=C\\:\\\path\\\hald.png[lut];[v][lut]haldclut]'
vf toggle lavfi='[null[v];movie=C\\:\\\\path\\\\hald.png[lut];[v][lut]haldclut]'
```

these are the two reference links i've found
https://stackoverflow.com/questions/60440793/how-can-i-use-windows-absolute-paths-with-the-movie-filter-on-ffmpeg
https://superuser.com/questions/1242652/is-it-possible-to-give-input-and-output-filess-absolute-path

also as a side note, is there a way to make my mpv always know a folder that is in my $PATH? that would be really helpful when applying luts or different types of overlays on the fly

**Reply by richardpl** [reply]
Not using Windows

**Reply by hooke007** [reply]
`vf toggle --vf=<filter name>` There is no such syntax.

`vf toggle <filter name>`

**Reply by junguler** [reply]
that's just a mistake i've made when copy pasting, you are correct tho, the issue still stands unfortunately but i will edit the post

**Reply by junguler** [reply]
my issue is this works
```mpv video.mp4 --vf=lavfi='[null[v];movie=c\\:\\\\path\\\\hald.png[lut];[v][lut]haldclut]'```
but it doesn't work as a keybind
```keybind vf toggle lavfi='[null[v];movie=c\\:\\\\path\\\\hald.png[lut];[v][lut]haldclut]'```
and i've tried with many other combinations of escaping the address but i can't find a solution to this at all, if the `hald.png` is not on the same folder as the video playing it will not work as a keybind, i guess i just have to do with the lut3d filter at the moment which works ...

**Reply by richardpl** [reply]
Why not using double quotes - "" ?

**Reply by junguler** [reply]
i've tried that already, it doesn't work unfortunately

**Reply by richardpl** [reply]
Its big c - C

**Reply by junguler** [reply]
tried that too, to no avail

**Comment by geextahslex** [comment]
You can try this glsl shader. It works without frei0r support

```
//!HOOK LUMA
//!BIND HOOKED

// Define the sRGB threshold and transition width
#define SRGB_THRESHOLD (170.0 / 255.0)
#define TRANSITION_WIDTH (10.0 / 255.0)

vec4 hook() {
    float luma = LUMA_texOff(0).x;

    // Use smoothstep for a smooth transition around the threshold
    float factor = smoothstep(SRGB_THRESHOLD - TRANSITION_WIDTH, SRGB_THRESHOLD + TRANSITION_WIDTH, luma);

    // Apply the inversion based on the smooth transition factor
    luma = mix(luma, 1.0 - luma, factor);

    return vec4(luma);
}
```

**Reply by junguler** [reply]
thanks for your reply, i ended up using lut3d which works but having an alternative is always nice

**Reply by geextahslex** [reply]
That sounds interesting, could you share a screenshot how this effect looks like?

**Comment by junguler** [comment]
![output](https://github.com/user-attachments/assets/206610e8-7c01-4b8e-9d87-12b165d614cc)

**Comment by junguler** [comment]
here is a simpler example

![output](https://github.com/user-attachments/assets/37e5279d-35a4-4491-9dc5-bd7754b2c0c4)
